lib/auth/v4/queryAuthCheck.js

```diff
@@ -35,6 +37,11 @@ queryAuthCheck.check = (request, log, callback) => {
     const expiry = authParams.expiry;
     const credential = authParams.credential;
 
+    if (!areSignedHeadersComplete(signedHeaders, request.headers)) {
+        log.debug('signedHeaders are incomplete', { signedHeaders });
+        return { err: errors.AccessDenied };
```

Comment:
Looks like a mistake, otherwise this needs some serious commenting/documenting, indeed.